pkg/pdapi/pdapi.go

```diff
@@ -67,6 +67,7 @@ func NewDefaultPDControl(kubeCli kubernetes.Interface) PDControlInterface {
 // It loads in-cluster root ca if caCert is empty.
 func GetTLSConfig(kubeCli kubernetes.Interface, namespace Namespace, tcName string, caCert []byte) (*tls.Config, error) {
 	secretName := util.ClusterClientTLSSecretName(tcName)
+	// TODO use secretLister instead of kubeCli.CoreV1().Secrets().Get() to optimize the performance
```

Comment:
this is intended, see https://github.com/pingcap/tidb-operator/issues/1327

Reply:
got it.